#47  added logging level to cookie, added unit test

dmzaytsev opened the pull request.

davvd commented:
Thanks, I will find someone to review this PR soon

davvd commented:
@pinaf please review, thanks

pinaf commented:
@dmzaytsev this needs a merge

dmzaytsev commented:
@pinaf my mistake
review it please

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -76,7 +77,14 @@ public RsFlash(final String msg, final Level level) {
+                        .append('/')
+                        .append(msg)
+                        .toString()
+                        .getBytes(Charset.defaultCharset())
```
@dmzaytsev isn't defaultcharset platform dependent? is fo, we might end up with different encodings depending on the platform. I suggest we fix this as utf-8.

dmzaytsev replied:
I also wanted to use UTF-8. But in many classes Take framework used default charset. I thought that there is reason for this.
I made #79 to clarify this

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -86,7 +94,5 @@ public RsFlash(final String msg, final Level level, final String cookie) {
             )
         )
         );
-        assert level != null;
```
@dmzaytsev i wonder why this line was here at all.

pinaf replied:
@dmzaytsev maybe we should add `@NotNull` to the `level` method parameter.

dmzaytsev replied:
I think the parameter had not been used before. To prevent the call constructor with null level

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -43,12 +46,25 @@
+                                .append('/')
+                                .append(msg)
+                                .toString()
+                                .getBytes(Charset.defaultCharset())
```
@dmzaytsev same utf-8 idea here.

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -43,12 +46,25 @@
+                            new StringBuilder(
+                                Level.INFO.getName()
+                            )
+                                .append('/')
```
@dmzaytsev not sure about the indentation here - maybe move this line up along with closing `)`. @yegor256 , what do you say?

yegor256 replied:
I would do `new StringBuilder(Level.INFO.getName())` in one line

pinaf commented:
@dmzaytsev a few comments above.

dmzaytsev commented:
@pinaf comments fixed
review again please

pinaf commented:
@dmzaytsev can't we add a `@NotNull` annotation to the level parameter?

yegor256 commented:
@pinaf we're not using Java validation annotations in this project

pinaf commented:
should we keep the null check then?

pinaf commented:
@rultor merge

rultor commented:
> @rultor merge

@pinaf Thanks for your request. @yegor256 Please confirm this.

yegor256 commented:
@dmzaytsev if `RsFlash` is encoding the text, we have to make sure `TsFlash` (and `TkFlash`) is decoding it

dmzaytsev commented:
@yegor256 I can't find TkFlash. Maybe do you mean  XeFlash ?

pinaf commented:
@yegor256 are you suggesting the creation of a puzzle?

yegor256 commented:
@dmzaytsev I meant `TsFlash` only (my bad). This class is mapping an incoming cookie to a header. We have to decode the cookie here if we encode it in `RsFlash`. Otherwise the entire mechanism just breaks. Would be great to test them together..

dmzaytsev commented:
@yegor256 Is the flash message should not be encoded in the the header?
The message may contain unprintable characters.
Or we need to make some processing with flash message instead simple translation the cookie to the header?
For example, XeFlash  break the flash to the level and the text. 
I guess XeFlash realy need changes, TsFlash no need. What do you think?

yegor256 commented:
@dmzaytsev you're absolutely right, again my bad (maybe this flash managing mechanism deserves a proper explanation in Javadoc... feel free to submit a bug about it). Yes, `XeFlash` has to unwrap the text and base64 decode it

dmzaytsev commented:
@pinaf please review, thanks

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -128,7 +138,5 @@ public RsFlash(final String msg, final Level level, final String cookie) {
             )
```
@dmzaytsev  this looks misaligned. it should close String.format

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -118,9 +120,17 @@ public RsFlash(final String msg, final Level level) {
+                        .append(level.getName())
+                        .toString()
+                        .getBytes(Charset.defaultCharset())
+            ),
```
@dmzaytsev this is misaligned. it should close `printBase64Binary`

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -128,7 +138,5 @@ public RsFlash(final String msg, final Level level, final String cookie) {
             )
         )
```
@dmzaytsev and this should align to `RsWithCookie`

pinaf commented on src/main/java/org/takes/facets/flash/XeFlash.java:
```diff
@@ -37,13 +40,18 @@
  * @since 0.1
  */
-@EqualsAndHashCode(of = { "req", "cookie" })
+@EqualsAndHashCode(of = {"req", "cookie" })
```
@dmzaytsev let's keep that whitespace

pinaf replied:
@dmzaytsev please re-add the whitespace

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -43,12 +46,23 @@
+                                .append(Level.INFO.getName())
+                                .toString()
+                                .getBytes(Charset.defaultCharset())
+                    )
```
@dmzaytsev align this to `DatatypeConverter.printBase64Binary`

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -43,12 +46,23 @@
+                                .toString()
+                                .getBytes(Charset.defaultCharset())
+                    )
+                ).toString()
```
@dmzaytsev align this to `append`

pinaf commented on src/test/java/org/takes/facets/flash/TsFlashTest.java:
```diff
@@ -71,7 +74,17 @@ public Take route(final Request request) {
+                                        .append(Level.INFO.getName())
+                                        .toString()
+                                        .getBytes(Charset.defaultCharset())
+                            )
```
@dmzaytsev align this to `DatatypeConverter.printBase64Binary`

pinaf commented on src/test/java/org/takes/facets/flash/TsFlashTest.java:
```diff
@@ -71,7 +74,17 @@ public Take route(final Request request) {
+                                        .toString()
+                                        .getBytes(Charset.defaultCharset())
+                            )
+                        ).toString()
```
@dmzaytsev align this to `append`

pinaf commented:
@dmzaytsev a few comments above, mostly about alignment of parentheses.

dmzaytsev commented:
@pinaf I agree with you regarding bracket's indentation, but checkstyle thinks otherwise

dmzaytsev commented:
I made commit  5d6a9cb with some  fixes. It won't pass CI checks

pinaf commented:
@dmzaytsev you are correct.. no idea why it complains. @yegor256 is this a problem with the qulice check or are we misapplying the rule?

yegor256 commented:
@dmzaytsev `.append(` should stay on the previous line, would look way nicer and checkstyle won't complain:

```
new StringBuilder("Cookie: RsFlash=").append(
  DatatypeConverter.printBase64Binary(
    new StringBuilder("Hello!")
      .append('/')
      .append(Level.INFO.getName())
      .toString()
      .getBytes(Charset.defaultCharset())
    )
).toString()
```

BTW, why do we need Base64 at all? We can use plain simple URLEncode, it will be enough

dmzaytsev commented:
@yegor256 I should add throws UnsupportedEncodingException in constructor in this case.
It this acceptable?

yegor256 commented:
@dmzaytsev IOException is acceptable, which is the same

dmzaytsev commented:
@pinaf review again please

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -32,10 +36,10 @@
 /**
  * Forwarding response.
- *
+ * <p/>
```
@dmzaytsev why all these paragraph tags?

dmzaytsev replied:
@pinaf I upgraded IntelliJ Idea, something wrong happened with code autoformat.

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -86,49 +90,69 @@
 
     /**
      * Ctor.
+     *
```
@dmzaytsev no need for these blank lines in method javadocs.

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -86,49 +90,69 @@
      * @param level Level
      * @param cookie Cookie name
+     * @throws IOException If fails
+     * @checkstyle IndentationCheck (28 lines)
```
@dmzaytsev we shouldn't have this. please remove it.

pinaf commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -86,49 +90,69 @@
-                cookie, msg,
+                cookie,
+                new StringBuilder(
+                        URLEncoder.encode(
```
@dmzaytsev the indentation here seems wrong (too much).

pinaf commented on src/main/java/org/takes/facets/flash/XeFlash.java:
```diff
@@ -65,6 +75,7 @@ public XeFlash(final Request request) {
 
     /**
      * Ctor.
+     *
```
@dmzaytsev same here. no need for blank lines in method javadocs.

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -39,16 +43,31 @@
 
     /**
      * RsFlash can add cookies.
+     *
```
@dmzaytsev same here. remove please.

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -39,16 +43,31 @@
      * RsFlash can add cookies.
+     *
      * @throws IOException If some problem inside
+     * @checkstyle IndentationCheck (24 lines)
```
@dmzaytsev we shouldn't have this. please remove it.

pinaf commented on src/test/java/org/takes/facets/flash/RsFlashTest.java:
```diff
@@ -39,16 +43,31 @@
-            Matchers.containsString("Set-Cookie: RsFlash=hey, how are you?")
+            Matchers.containsString(
+                new StringBuilder("Set-Cookie: RsFlash=")
+                    .append(
```
@dmzaytsev can we move this append to the line above?

pinaf commented on src/test/java/org/takes/facets/flash/TsFlashTest.java:
```diff
@@ -23,26 +23,23 @@
  * @version $Id$
  * @since 0.4
- * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @todo The test must be rewritten because
```
@dmzaytsev what happened here?

pinaf commented on src/test/java/org/takes/facets/flash/TsFlashTest.java:
```diff
@@ -56,30 +53,18 @@ public void parsesCookieIntoFlash() throws IOException {
-                            )
-                        )
-                    );
+                    return new TkEmpty();
```
@dmzaytsev this isn't testing anything. what happened?

pinaf commented on src/test/java/org/takes/facets/flash/XeFlashTest.java:
```diff
@@ -42,11 +49,24 @@
     /**
-     * XeFlash can create a flash data.
+     * XeFlash can accept RsFlash cookie.
+     *
```
@dmzaytsev remove this line please

pinaf commented:
@dmzaytsev multiple comments above.

dmzaytsev commented:
@pinaf Could you suggest please how to reformat this code ? Checkstyle complains.
```
        super(
            new RsWithCookie(
                cookie,
                String.format(
                    "%s/%s",
                    URLEncoder.encode(msg, Charset.defaultCharset().name()),
                    URLEncoder.encode(
                        level.getName(),
                        Charset.defaultCharset().name()
                    )
                ),
                "Path=/",
                String.format(
                    Locale.ENGLISH,
                    "Expires=%1$ta, %1$td %1$tb %1$tY %1$tT GMT",
                    new Date(
                        System.currentTimeMillis() + TimeUnit.HOURS.toMillis(1L)
                    )
                )
            )
        );
```

yegor256 commented:
@dmzaytsev it's a bug in Checkstyle, just do what it says and shift the closing bracket left

pinaf commented:
@rultor merge

rultor commented:
> @rultor merge

@pinaf Thanks for your request. @yegor256 Please confirm this.

yegor256 commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -74,9 +78,8 @@
  * default name using a constructor of {@link org.takes.facets.flash.RsFlash},
  * but it's not recommended. It's better to use the default name.
- *
+ * <p/>
```
what this is for?

dmzaytsev replied:
IntelliJ Idea code format feature...
I'll deal with it

yegor256 commented on src/main/java/org/takes/facets/flash/RsFlash.java:
```diff
@@ -114,13 +120,24 @@ public RsFlash(final String msg, final Level level) {
+                String.format(
+                    "%s/%s",
+                    URLEncoder.encode(msg, Charset.defaultCharset().name()),
+                    URLEncoder.encode(
```
I don't think we need to URL-encode logging level. it's always one English word

yegor256 commented on src/main/java/org/takes/facets/flash/XeFlash.java:
```diff
@@ -37,13 +40,18 @@
+    /**
+     * Compiled RsFlash message regexp pattern.
+     */
+    private static final Pattern RS_FLASH_MSG = Pattern.compile(
```
isn't it easier/faster to use `String.split()` instead?

dmzaytsev replied:
you're right, it was done before schema encoding changes

dmzaytsev replied:
my bad, previous comment wrong
I think compiled regexp will be faster

yegor256 commented:
@dmzaytsev see my comments above

dmzaytsev commented:
@yegor256 I fixed all comments, except [this](https://github.com/yegor256/takes/pull/78#discussion-diff-27421305). I think compiled regexp will be faster.
If you really would like to use String.split (), I will do it.

yegor256 commented:
@rultor try to merge it

rultor commented:
> @rultor try to merge it

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/2876-87949921)

rultor commented:
> @rultor try to merge it

@yegor256 Oops, I failed. You can see the full log [here](http://www.rultor.com/t/2876-87949921) (spent 1min)

```
+ set -e
+ set -o pipefail
++ dirname /tmp/rultor-P1HC/run.sh
+ cd /tmp/rultor-P1HC
+ echo 3853
+ echo '2.0-SNAPSHOT BUILD'
2.0-SNAPSHOT BUILD
+ date
Tue Mar 31 08:16:46 CEST 2015
+ uptime
 08:16:46 up 43 days,  9:47,  0 users,  load average: 10.62, 11.71, 9.89
+ fork=git@github.com:dmzaytsev/takes.git
+ fork_branch=47
+ head=git@github.com:yegor256/takes.git
+ head_branch=master
+ pull_id=78
+ vars=('--env=MAVEN_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=JAVA_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=fork=git@github.com:dmzaytsev/takes.git' '--env=fork_branch=47' '--env=head=git@github.com:yegor256/takes.git' '--env=head_branch=master' '--env=pull_id=78')
+ image=yegor256/rultor
+ squash=false
+ directory=
+ scripts=('sudo locale-gen en_US en_US.UTF-8' ';' 'sudo dpkg-reconfigure locales' ';' 'export LC_ALL=en_US.UTF-8' ';' 'export LANG=en_US.UTF-8' ';' 'export LANGUAGE=en_US.UTF-8' ';' 'sudo gem install pdd' ';' 'version=$(curl -K ../curl-appveyor.cfg --data "{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''${pull_id}'\''}" https://ci.appveyor.com/api/builds | jq -r '\''.version'\'')' ';' 'while true' ';' 'do status=$(curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/${version} | jq -r '\''.build.status'\'')' ';' 'if [ "${status}" == "success" ]; then break; fi' ';' 'if [ "${status}" == "failed" ]; then echo "see https://ci.appveyor.com/project/yegor256/takes/build/${version}"; exit 1; fi' ';' 'echo "waiting for AppVeyor build ${version}: ${status}"' ';' 'sleep 5s' ';' 'done' ';' 'mvn clean install -Ptakes -Pqulice --errors --settings ../settings.xml' ';' 'mvn clean site -Psite --errors --settings ../settings.xml' ';' 'mvn clean' ';' 'pdd --source=$(pwd) --verbose --file=/dev/null' ';')
+ container=yegor256_takes_78
+ as_root=false
+ git clone --branch=master --depth=10 git@github.com:yegor256/takes.git repo
Initialized empty Git repository in /tmp/rultor-P1HC/repo/.git/
+ cd repo
+ git config user.email me@rultor.com
+ git config user.name rultor
+ '[' -z 'sudo locale-gen en_US en_US.UTF-8' ']'
+ cd ..
+ cat
+ '[' false = true ']'
+ cat
+ chmod a+x entry.sh
+ cat
+ echo 'sudo locale-gen en_US en_US.UTF-8' ';' 'sudo dpkg-reconfigure locales' ';' 'export LC_ALL=en_US.UTF-8' ';' 'export LANG=en_US.UTF-8' ';' 'export LANGUAGE=en_US.UTF-8' ';' 'sudo gem install pdd' ';' 'version=$(curl -K ../curl-appveyor.cfg --data "{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''${pull_id}'\''}" https://ci.appveyor.com/api/builds | jq -r '\''.version'\'')' ';' 'while true' ';' 'do status=$(curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/${version} | jq -r '\''.build.status'\'')' ';' 'if [ "${status}" == "success" ]; then break; fi' ';' 'if [ "${status}" == "failed" ]; then echo "see https://ci.appveyor.com/project/yegor256/takes/build/${version}"; exit 1; fi' ';' 'echo "waiting for AppVeyor build ${version}: ${status}"' ';' 'sleep 5s' ';' done ';' 'mvn clean install -Ptakes -Pqulice --errors --settings ../settings.xml' ';' 'mvn clean site -Psite --errors --settings ../settings.xml' ';' 'mvn clean' ';' 'pdd --source=$(pwd) --verbose --file=/dev/null' ';'
+ gpg --keyserver hkp://pool.sks-keyservers.net --verbose --recv-keys 9AF0FA4C
gpg: requesting key 9AF0FA4C from hkp server pool.sks-keyservers.net
Version: SKS 1.1.5
gpg: armor header: 
Comment: Hostname: keyserver.durcheinandertal.ch
gpg: armor header: 
gpg: pub  2048R/9AF0FA4C 2014-08-17  Rultor.com <gpg@rultor.com>
gpg: key 9AF0FA4C: "Rultor.com <gpg@rultor.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
++ pwd
++ pwd
+ gpg --verbose --keyring=/tmp/rultor-P1HC/.gpg/pubring.gpg --secret-keyring=/tmp/rultor-P1HC/.gpg/secring.gpg --decrypt repo/curl-appveyor.cfg.asc
Version: GnuPG v1
gpg: armor header: 
gpg: public key is 9A839C3A
gpg: using subkey 9A839C3A instead of primary key 9AF0FA4C
gpg: using subkey 9A839C3A instead of primary key 9AF0FA4C
gpg: encrypted with 2048-bit RSA key, ID 9A839C3A, created 2014-08-17
      "Rultor.com <gpg@rultor.com>"
gpg: AES256 encrypted data
gpg: original file name='curl-appveyor.cfg.enc'
+ gpg --no-tty --batch --verbose --decrypt --passphrase rultor-key:yegor256/takes curl-appveyor.cfg.enc
Version: GnuPG v1
gpg: armor header: 
gpg: CAST5 encrypted data
gpg: encrypted with 1 passphrase
gpg: original file name='curl-appveyor.cfg'
gpg: WARNING: message was not integrity protected
+ rm -rf curl-appveyor.cfg.enc
+ rm -rf .gpg
+ cd repo
+ git remote add fork git@github.com:dmzaytsev/takes.git
+ git remote update
Fetching origin
Fetching fork
From github.com:dmzaytsev/takes
 * [new branch]      28         -> fork/28
 * [new branch]      36         -> fork/36
 * [new branch]      47         -> fork/47
 * [new branch]      74         -> fork/74
 * [new branch]      gh-pages   -> fork/gh-pages
 * [new branch]      master     -> fork/master
+ '[' false == true ']'
+ git merge fork/47
Auto-merging .rultor.yml
Auto-merging README.md
Auto-merging curl-appveyor.cfg.asc
CONFLICT (add/add): Merge conflict in curl-appveyor.cfg.asc
Auto-merging src/main/java/org/takes/NotFoundException.java
Auto-merging src/main/java/org/takes/facets/flash/RsFlash.java
Auto-merging src/main/java/org/takes/facets/flash/TsFlash.java
Auto-merging src/main/java/org/takes/facets/flash/XeFlash.java
CONFLICT (content): Merge conflict in src/main/java/org/takes/facets/flash/XeFlash.java
Auto-merging src/main/java/org/takes/rs/RsGzip.java
CONFLICT (add/add): Merge conflict in src/main/java/org/takes/rs/RsGzip.java
Auto-merging src/test/java/org/takes/facets/flash/XeFlashTest.java
CONFLICT (content): Merge conflict in src/test/java/org/takes/facets/flash/XeFlashTest.java
Auto-merging src/test/java/org/takes/rs/RsGzipTest.java
CONFLICT (add/add): Merge conflict in src/test/java/org/takes/rs/RsGzipTest.java
Automatic merge failed; fix conflicts and then commit the result.

```

dmzaytsev commented:
@yegor256 I tried to merge 47 with upstream/master, 
 src/main/java/org/takes/facets/flash/XeFlash.java not changed
other files from rultor's log not exists in my commit

yegor256 commented:
@rultor try to merge again

dmzaytsev commented:
@yegor256 seems rultor didn't merge

rultor commented:
> @rultor try to merge again

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/2876-88213522)

rultor commented:
> @rultor try to merge again

@yegor256 Oops, I failed. You can see the full log [here](http://www.rultor.com/t/2876-88213522) (spent 1min)

```
+ set -e
+ set -o pipefail
++ dirname /tmp/rultor-oxGJ/run.sh
+ cd /tmp/rultor-oxGJ
+ echo 1655
+ echo '1.52.2 2203696'
1.52.2 2203696
+ date
Wed Apr  1 07:41:53 CEST 2015
+ uptime
 07:41:53 up 44 days,  9:13,  0 users,  load average: 7.69, 8.30, 7.38
+ fork=git@github.com:dmzaytsev/takes.git
+ fork_branch=47
+ head=git@github.com:yegor256/takes.git
+ head_branch=master
+ pull_id=78
+ vars=('--env=MAVEN_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=JAVA_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=fork=git@github.com:dmzaytsev/takes.git' '--env=fork_branch=47' '--env=head=git@github.com:yegor256/takes.git' '--env=head_branch=master' '--env=pull_id=78')
+ image=yegor256/rultor
+ squash=false
+ directory=
+ scripts=('sudo locale-gen en_US en_US.UTF-8' ';' 'sudo dpkg-reconfigure locales' ';' 'export LC_ALL=en_US.UTF-8' ';' 'export LANG=en_US.UTF-8' ';' 'export LANGUAGE=en_US.UTF-8' ';' 'sudo gem install pdd' ';' 'version=$(curl -K ../curl-appveyor.cfg --data "{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''${pull_id}'\''}" https://ci.appveyor.com/api/builds | jq -r '\''.version'\'')' ';' 'while true' ';' 'do status=$(curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/${version} | jq -r '\''.build.status'\'')' ';' 'if [ "${status}" == "success" ]; then break; fi' ';' 'if [ "${status}" == "failed" ]; then echo "see https://ci.appveyor.com/project/yegor256/takes/build/${version}"; exit 1; fi' ';' 'echo "waiting for AppVeyor build ${version}: ${status}"' ';' 'sleep 5s' ';' 'done' ';' 'mvn clean install -Ptakes -Pqulice --errors --settings ../settings.xml' ';' 'mvn clean site -Psite --errors --settings ../settings.xml' ';' 'mvn clean' ';' 'pdd --source=$(pwd) --verbose --file=/dev/null' ';')
+ container=yegor256_takes_78
+ as_root=false
+ git clone --branch=master --depth=10 git@github.com:yegor256/takes.git repo
Initialized empty Git repository in /tmp/rultor-oxGJ/repo/.git/
+ cd repo
+ git config user.email me@rultor.com
+ git config user.name rultor
+ '[' -z 'sudo locale-gen en_US en_US.UTF-8' ']'
+ cd ..
+ cat
+ '[' false = true ']'
+ cat
+ chmod a+x entry.sh
+ cat
+ echo 'sudo locale-gen en_US en_US.UTF-8' ';' 'sudo dpkg-reconfigure locales' ';' 'export LC_ALL=en_US.UTF-8' ';' 'export LANG=en_US.UTF-8' ';' 'export LANGUAGE=en_US.UTF-8' ';' 'sudo gem install pdd' ';' 'version=$(curl -K ../curl-appveyor.cfg --data "{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''${pull_id}'\''}" https://ci.appveyor.com/api/builds | jq -r '\''.version'\'')' ';' 'while true' ';' 'do status=$(curl -K ../curl-appveyor.cfg https://ci.appveyor.com/api/projects/yegor256/takes/build/${version} | jq -r '\''.build.status'\'')' ';' 'if [ "${status}" == "success" ]; then break; fi' ';' 'if [ "${status}" == "failed" ]; then echo "see https://ci.appveyor.com/project/yegor256/takes/build/${version}"; exit 1; fi' ';' 'echo "waiting for AppVeyor build ${version}: ${status}"' ';' 'sleep 5s' ';' done ';' 'mvn clean install -Ptakes -Pqulice --errors --settings ../settings.xml' ';' 'mvn clean site -Psite --errors --settings ../settings.xml' ';' 'mvn clean' ';' 'pdd --source=$(pwd) --verbose --file=/dev/null' ';'
+ gpg --keyserver hkp://pool.sks-keyservers.net --verbose --recv-keys 9AF0FA4C
gpg: requesting key 9AF0FA4C from hkp server pool.sks-keyservers.net
Version: SKS 1.1.5
gpg: armor header: 
Comment: Hostname: keyserver.synack.com
gpg: armor header: 
gpg: pub  2048R/9AF0FA4C 2014-08-17  Rultor.com <gpg@rultor.com>
gpg: key 9AF0FA4C: "Rultor.com <gpg@rultor.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
++ pwd
++ pwd
+ gpg --verbose --keyring=/tmp/rultor-oxGJ/.gpg/pubring.gpg --secret-keyring=/tmp/rultor-oxGJ/.gpg/secring.gpg --decrypt repo/curl-appveyor.cfg.asc
Version: GnuPG v1
gpg: armor header: 
gpg: public key is 9A839C3A
gpg: using subkey 9A839C3A instead of primary key 9AF0FA4C
gpg: using subkey 9A839C3A instead of primary key 9AF0FA4C
gpg: encrypted with 2048-bit RSA key, ID 9A839C3A, created 2014-08-17
      "Rultor.com <gpg@rultor.com>"
gpg: AES256 encrypted data
gpg: original file name='curl-appveyor.cfg.enc'
+ gpg --no-tty --batch --verbose --decrypt --passphrase rultor-key:yegor256/takes curl-appveyor.cfg.enc
Version: GnuPG v1
gpg: armor header: 
gpg: CAST5 encrypted data
gpg: encrypted with 1 passphrase
gpg: original file name='curl-appveyor.cfg'
gpg: WARNING: message was not integrity protected
+ rm -rf curl-appveyor.cfg.enc
+ rm -rf .gpg
+ cd repo
+ git remote add fork git@github.com:dmzaytsev/takes.git
+ git remote update
Fetching origin
Fetching fork
From github.com:dmzaytsev/takes
 * [new branch]      28         -> fork/28
 * [new branch]      36         -> fork/36
 * [new branch]      47         -> fork/47
 * [new branch]      74         -> fork/74
 * [new branch]      gh-pages   -> fork/gh-pages
 * [new branch]      master     -> fork/master
+ '[' false == true ']'
+ git merge fork/47
Auto-merging .rultor.yml
Auto-merging README.md
Auto-merging src/main/java/org/takes/NotFoundException.java
Auto-merging src/main/java/org/takes/facets/flash/RsFlash.java
Auto-merging src/main/java/org/takes/facets/flash/TsFlash.java
Auto-merging src/main/java/org/takes/facets/flash/XeFlash.java
CONFLICT (content): Merge conflict in src/main/java/org/takes/facets/flash/XeFlash.java
Auto-merging src/test/java/org/takes/facets/flash/XeFlashTest.java
CONFLICT (content): Merge conflict in src/test/java/org/takes/facets/flash/XeFlashTest.java
Automatic merge failed; fix conflicts and then commit the result.

```

dmzaytsev commented:
@yegor256 
>This pull request can be automatically merged by project collaborators.

But Rultor reports about merge error. Is it my problem or Rultor  ? What I should to do ?
Let me know please

yegor256 commented:
@dmzaytsev just merge master branch into your branch and push:

```
git fetch
git checkout 47
git merge master
git push origin 47
```

dmzaytsev commented:
@yegor256 try to merge please

yegor256 commented:
@rultor merge it

rultor commented:
> @rultor merge it

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/2876-88719580)

rultor commented:
> @rultor merge it

@yegor256 Oops, I failed. You can see the full log [here](http://www.rultor.com/t/2876-88719580) (spent 3min)

```
6c37f792ddac: Download complete
e54ca5efa2e9: Download complete
18e5e08ce16d: Download complete
cfd3123899d7: Download complete
7fdbbba76594: Download complete
452bfe735d07: Download complete
2d8dc22e2c1a: Download complete
5dba824e2096: Download complete
996aaaaa3be6: Download complete
e2730cdac0e5: Download complete
3d8c369c7792: Download complete
2ed8d6c88527: Download complete
39a477af720b: Download complete
b97894feaf2c: Download complete
5a2f0c609499: Download complete
9ad8daa2cd1d: Download complete
fbb3c7ffb85b: Download complete
63892fa9dc21: Download complete
244638a429a4: Download complete
c0b899df8169: Download complete
689ff7a95ae2: Download complete
da4c3c51f876: Download complete
af8fbf8cd4dc: Download complete
7d0fc5b517c8: Download complete
a7280d0d7ae9: Download complete
2ad7249bf03c: Download complete
1ccd85a022a2: Download complete
ef7a01b3f761: Download complete
99499274d0d3: Download complete
df0608d44329: Download complete
df0608d44329: Download complete
Status: Image is up to date for yegor256/rultor:latest
+ docker ps --filter=status=exited
++ pwd
++ pwd
+ docker run --rm -v /tmp/rultor-fgiM:/main '--env=MAVEN_OPTS=-XX:MaxPermSize=256m -Xmx1g' '--env=JAVA_OPTS=-XX:MaxPermSize=256m -Xmx1g' --env=fork=git@github.com:dmzaytsev/takes.git --env=fork_branch=47 --env=head=git@github.com:yegor256/takes.git --env=head_branch=master --env=pull_id=78 --memory=4g --cidfile=/tmp/rultor-fgiM/cid -w=/main --name=yegor256_takes_78 yegor256/rultor /main/entry.sh
+ set -e
+ set -o pipefail
+ shopt -s dotglob
+ useradd -m -G sudo r
+ echo '%sudo ALL=(ALL) NOPASSWD:ALL'
+ cp -R /root/.bashrc /root/.ivy2 /root/.java /root/.m2 /root/.profile /root/.sbt /home/r
+ cp -R ./cid ./curl-appveyor.cfg ./entry.sh ./pid ./pubring.gpg ./repo ./run.sh ./s3cfg ./script.sh ./secring.gpg ./settings.xml ./stdout /home/r
+ rm -rf repo
+ chown -R r:r /home/r
+ chmod a+x /home/r/script.sh
+ su -m r -c /home/r/script.sh
+ set -e
+ set -o pipefail
+ export HOME=/home/r
+ HOME=/home/r
+ cd /home/r/repo
+ sudo locale-gen en_US en_US.UTF-8
Generating locales...
  en_US.ISO-8859-1... done
  en_US.UTF-8... done
Generation complete.
+ sudo dpkg-reconfigure locales
debconf: unable to initialize frontend: Dialog
debconf: (Dialog frontend will not work on a dumb terminal, an emacs shell buffer, or without a controlling terminal.)
debconf: falling back to frontend: Readline
debconf: unable to initialize frontend: Readline
debconf: (This frontend requires a controlling tty.)
debconf: falling back to frontend: Teletype
Generating locales...
  en_US.ISO-8859-1... up-to-date
  en_US.UTF-8... up-to-date
Generation complete.
+ export LC_ALL=en_US.UTF-8
+ LC_ALL=en_US.UTF-8
+ export LANG=en_US.UTF-8
+ LANG=en_US.UTF-8
+ export LANGUAGE=en_US.UTF-8
+ LANGUAGE=en_US.UTF-8
+ sudo gem install pdd
Building native extensions.  This could take a while...
Building native extensions.  This could take a while...

ruby-filemagic-0.6.1 [2014-08-19]:

* Account for options passed to FileMagic.fm (issue #7 reported by Adam
  Wróbel).

Successfully installed nokogiri-1.6.5
Successfully installed ruby-filemagic-0.6.1
Successfully installed rake-10.4.2
Successfully installed pdd-0.14
4 gems installed
Installing ri documentation for nokogiri-1.6.5...
Installing ri documentation for ruby-filemagic-0.6.1...
Installing ri documentation for rake-10.4.2...
Installing ri documentation for pdd-0.14...
Installing RDoc documentation for nokogiri-1.6.5...
Installing RDoc documentation for ruby-filemagic-0.6.1...
Installing RDoc documentation for rake-10.4.2...
Installing RDoc documentation for pdd-0.14...
++ curl -K ../curl-appveyor.cfg --data '{accountName: '\''yegor256'\'', projectSlug: '\''takes'\'', pullRequestId: '\''78'\''}' https://ci.appveyor.com/api/builds
++ jq -r .version
+ version=

```

pinaf commented:
@rultor merge

rultor commented:
> @rultor merge

@pinaf Thanks for your request. @yegor256 Please confirm this.

yegor256 commented:
@rultor merge

rultor commented:
> @rultor merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/2876-88744548)

rultor commented:
> @rultor merge

@yegor256 Done! FYI, the full log is [here](http://www.rultor.com/t/2876-88744548) (took me 18min)

davvd commented:
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

davvd commented:
@rultor deploy

rultor commented:
> @rultor deploy

@davvd OK, I'll try to deploy now. You can check the progress [here](http://www.rultor.com/t/2876-89359133)

rultor commented:
> @rultor deploy

@davvd Done! FYI, the full log is [here](http://www.rultor.com/t/2876-89359133) (took me 14min)

elenavolokhova commented:
@davvd This ticket looks good! :)

davvd commented:
> @davvd This ticket looks good! :)

@elenavolokhova thanks, got it

davvd commented:
@pinaf just added 10 mins to @elenavolokhova (for QA), payment ID is `54704853`; Thanks a lot, I just topped your account for **26 mins**, transaction ID `AP-36U20597B5681625G`, total time was 162 hours and 42 mins; review comments (c=11) added as a bonus; +26 added to your rating, at the moment it is: [+6080](http://www.netbout.com/b/35128?open=rating)
